akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -510,6 +518,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var shortCircuitBuffer: ByteBuffer[BoundaryEvent] = _
+  private var currentLimit: Int = shortCircuitLimit
```

Comment:
@agolubev so `syncProcessingLimit` is isomorphic to `throughput` for the dispatcher? Then I'd reuse that setting, or derive something from it. Because the problem is that the only way you can make an informed decision about it is if you know everything that is executing on the same threadpool.